pkg/skaffold/build/custom/custom.go

```diff
@@ -89,20 +89,27 @@ func (b *ArtifactBuilder) retrieveCmd(out io.Writer, a *latest.Artifact, tag str
 	return cmd, nil
 }
 
-func (b *ArtifactBuilder) retrieveEnv(a *latest.Artifact, tag string) ([]string, error) {
-	buildContext, err := buildContext(a.Workspace)
+func (b *ArtifactBuilder) retrieveEnv(workspace string, a *latest.CustomArtifact, tag string) ([]string, error) {
+	buildContext, err := buildContext(workspace)
 	if err != nil {
 		return nil, errors.Wrap(err, "getting absolute path for artifact build context")
 	}
 
-	envs := []string{
+	evaluatedEnv, err := misc.EvaluateEnv(a.Env)
+	if err != nil {
+		return nil, errors.Wrap(err, "unable to evaluate env variables")
+	}
+
+	var envs []string
+	envs = append(envs, util.OSEnviron()...)
+	envs = append(envs, b.additionalEnv...)
+	envs = append(envs, evaluatedEnv...)
```

Comment:
Is there an implied order here?  I'd think that later values will override earlier values in the array, such that this means that values in the `skaffold.yaml` override those on the command-line?  Which seems unusual.

(I wonder if this function might not be better converting to a map and then back to a string array?)